File: 23-3-2024_Fibonacci series up to Nth term.java
//{ Driver Code Starts
import java.io.*;
import java.util.*;

class Main {
    // Driver code
    public static void main(String[] args) throws Exception {
        BufferedReader br =
            new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(br.readLine().trim());
        while (t-- > 0) {
            int n = Integer.parseInt(br.readLine().trim());
            Solution obj = new Solution();
            int ans[] = obj.Series(n);
            for (int i : ans) {
                System.out.print(i + " ");
            }
            System.out.println();
        }
    }
}
// } Driver Code Ends


class Solution {

    int[] Series(int n) {
        // code here
        int mod=(int)1e9+7;
        int[] ans=new int[n+1];
        ans[1]=1;
        for(int i=2;i<=n;i++){
            ans[i]=((ans[i-1]%mod)+(ans[i-2]%mod))%mod;
        }
        return ans;
    }
}

File: 23-3-2024_Fibonacci series up to Nth term.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;

// } Driver Code Ends
class Solution {
  public:
    vector<int> Series(int n) {
        // Code here
         vector<int>ans;
        long long a=0;
        long long b=1;
        ans.push_back(a);
        ans.push_back(b);
        long long mod=1e9+7;
        for(int i=2;i<=n;i++){
            long long sum=(a+b)%mod;
            a=b;
            b=sum;
            ans.push_back(sum);
        }
        return ans;
    }
    };

//{ Driver Code Starts.
int main() {
    int t;
    cin >> t;
    while (t--) {
        int n;
        cin >> n;
        Solution obj;

        vector<int> ans = obj.Series(n);
        for (auto x : ans) cout << x << " ";
        cout << "\n";
    }
    return 0;
}
// } Driver Code Ends